app/src/main/java/com/rockspin/androiddevtest/EVActivityAdapter.java

```diff
@@ -40,9 +45,17 @@ public ViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
 
     @Override
     public void onBindViewHolder(ViewHolder holder, int position) {
-        final CosmonautActivity cosmonautActivity = mCosmonautActivityList.get(position);
+
+        int index   = position;
+        if( ! m_ascending )
+            index   = mCosmonautActivityList.size() - 1 - position;
```

Comment:
Interesting list way of sorting the list. Why not just reorder the array? and notify the adapter. This will only work if the server returns results predictably. What happens if the server changes ordering?